Extensions/XMPPMUCSUB/XMPPMUCSUB.h

```diff
@@ -0,0 +1,264 @@
+ * In order to change the nick and/or subscription nodes, the same request MUST be sent with a different nick or nodes information.
+ *
+ * A room moderator can subscribe another user to MUC Room events by providing the user JID as an attribute in the <subscribe/> element.
+ *
+ * @see xmppMUCSUB:didSubscribeForRoomJID:
+ * @see xmppMUCSUB:failedToSubscribeForRoomJID:withError:
+ *
+ * @param events the events that user wants to subscribe to.
+ * @param roomJID the room that user wants to subscribe to.
+ * @param userJID Room moderator subscribes another room user.
+ * @param nickName nickname of user in room.
+ * @param password room password if there is any.
+ */
+- (void)subscibeToEvents:(NSArray<XMPPSubscribeEvent> *)events roomJID:(XMPPJID *)roomJID userJID:(XMPPJID *)userJID withNick:(NSString *)nickName passwordForRoom:(NSString *)password;
+
+/**
+ * User can unsubscribe from MUC Room events
+ *
+ * A room moderator can unsubscribe another room user from MUC Room events by providing the user JID as an attribute in the <unsubscribe/> element.
+ *
+ * @see xmppMUCSUB:didUnsubscribeFromRoomJID:
+ * @see xmppMUCSUB:failedToUnsubscribeFromRoomJID:withError:
+ *
+ * @param roomJID the room that user wants to subscribe to.
+ * @param userJID Room moderator unsubscribes another room user.
+ */
+- (void)unsubscibeFromRoomJID:(XMPPJID *)roomJID userJID:(XMPPJID *)userJID;
+
+/**
+ * A user can query the MUC service to get their list of subscriptions.
+ *
+ */
+- (void)fetchSubscriptionList;
+
+/**
+ * A room moderator can get the list of subscribers by sending <subscriptions/> request directly to the room JID.
+ *
+ * @param roomJID the room that user wants to get list of subscribers.
+ */
+- (void)fetchSubscribersListForRoom:(XMPPJID *)roomJID;
+
+@end
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+#pragma mark -
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+@protocol XMPPMUCSUBDelegate
+@optional
+
+- (void)xmppMUCSUB:(XMPPMUCSUB *)sender didReceiveMessage:(XMPPMessage *)message;
+- (void)xmppMUCSUB:(XMPPMUCSUB *)sender didReceivePresence:(XMPPPresence *)message;
+- (void)xmppMUCSUB:(XMPPMUCSUB *)sender didSubscribe:(NSXMLElement *)subscribe;
+- (void)xmppMUCSUB:(XMPPMUCSUB *)sender didUnsubscribe:(NSXMLElement *)unsubscribe;
+
+/**
+ * Implement this method when calling [mucsubInstanse discoverFeatures]. It will be invoked if the request
+ * for discovering services is successfully executed and receives a successful response.
+ *
+ * @param sender XMPPMUCSUB object invoking this delegate method.
+ * @param features An array of NSXMLElements in the form shown below. You will need to extract the data you
+ *                 wish to use.
+ *
+ *                 <feature var="urn:xmpp:mucsub:0" />
+ */
+- (void)xmppMUCSUB:(XMPPMUCSUB *)sender didDiscoverFeatures:(NSArray *)features;
```

Comment:
Can we add <ObjectType *> to the array definition?

Reply:
Done